app/policies/mixins/rbac_mixin.rb

```diff
@@ -34,58 +37,28 @@ def requester?(klass = @record.class, verb = 'read')
       scopes(klass, verb).include?("user")
     end
 
-    # check if approver can process the #{resource} with #{id}
-    def approvable?(resource, id)
-      approver_id_list(resource)&.include?(id.to_i)
-    end
-
-    # check if regular requester own the #{resource} with #{id}
-    def owned?(resource, id)
-      owner_id_list(resource)&.include?(id.to_i)
+    def approver_visible_requests(scope)
+      scope.where(:group_ref => @user.group_uuids, :state => APPROVER_VISIBLE_STATES)
     end
 
-    # resource ids approver can access
-    def approver_id_list(resource)
-      visible_request_ids = visible_request_ids_for_approver
-      Rails.logger.debug { "Final accessible request ids: #{visible_request_ids}" }
-
-      case resource
-      when "requests"
-        visible_request_ids
-      when "actions"
-        Action.where(:request_id => visible_request_ids).pluck(:id).sort
-      else
-        raise ArgumentError, "Unknown resource type: #{resource}"
-      end
+    def scopes(klass, verb)
+      access.scopes(klass.table_name, verb)
     end
 
-    # resource ids owner owns
-    def owner_id_list(resource)
-      case resource
-      when "requests"
-        owner_request_ids
-      when "actions"
-        Action.where(:request_id => owner_request_ids).pluck(:id).sort
-      else
-        raise ArgumentError, "Unknown resource type: #{resource}"
-      end
+    def access
+      @user.access
     end
 
-    def owner_request_ids
-      Request.by_owner.pluck(:id).sort
-    end
+    private
 
-    # All child request ids for approver to process
-    def visible_request_ids_for_approver
-      Request.where(:group_ref => @user.group_uuids, :state => APPROVER_VISIBLE_STATES).pluck(:id)
+    def approver_accessible?(record)
```

Comment:
Approver can only read the requests whose states in `APPROVER_VISIBLE_STATES`. This check is missed here.